### ISSUE TITLE
Spatial Cross Entropy Loss - Feature Request

### ISSUE THREAD

**Issue by kvrd18** [tensorflow/tensorflow#7156]
For Fully Convolutional Networks, the logits are reshaped into 2D tensors `[batch_size, n_classes]`. Can't there be a spatial loss function similar to [SpatialClassNLLCriterion](https://github.com/torch/cunn/blob/master/lib/THCUNN/SpatialClassNLLCriterion.cu') built by Torch?

**Comment by girving** [comment]
Your link to torch is broken.

**Comment by kvrd18** [comment]
Try [this](https://github.com/torch/cunn/blob/master/lib/THCUNN/SpatialClassNLLCriterion.cu) link?

**Comment by girving** [comment]
Ah, it's just a link to code, not documentation.  Could you give a self contained mathematical description?

**Comment by girving** [comment]
Closing for now, but I'm happy to reopen if there is a mathematical description that seems like something useful.  I'm not going to reverse engineer the meaning from code.

**Comment by kvrd18** [comment]
I figured that [`tf.nn.softmax_cross_entropy_with_logits`](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) computes the cross entropy map even for spatial inputs and spatial targets. This issue can be considered closed. Thanks!

**Comment by techmatt** [comment]
tf.nn.sparse_softmax_cross_entropy_with_logits is the equivalent function here -- the "sparse_" means that the distribution is hard, i.e. (0, 1, 0, ..., 0) not (0.5, 0.25, 0.25, 0, ... 0), which is what SpatialClassNLLCriterion is also assumes
